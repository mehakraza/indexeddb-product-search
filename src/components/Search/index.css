.search {
  min-height: 72px;
  padding-top: 20px;
}