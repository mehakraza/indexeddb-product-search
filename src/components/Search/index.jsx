import React from 'react';
import { Row, Col, Spin, Input, Select, Checkbox } from 'antd';
import './index.css';

const GENDERS = ['male', 'female', 'unisex'];

const Search = ({ searchRef, searching, setSearchTerm, setGender, setOnSale, searchTerm, gender, onSale }) => {
  const handleOnSaleChange = e => {
    const newOnSale = e.target.checked;
    setOnSale(newOnSale);
  };

  const handleSearchChange = e => {
    if (e.key === 'Enter') return;

    setSearchTerm(`${searchTerm}${e.key}`);
  }

  const searchBar = (
    <div className="search">
      <Row align="middle" justify="space-around" gutter={16}>
        <Col offset={2} span={16}>
          <Input
            placeholder="Search by title"
            onKeyPress={handleSearchChange}
            onChange={e => { if (!e.target.value) setSearchTerm('') }}
            value={searchTerm}
            allowClear={true}
            ref={searchRef}
          />
        </Col>
        <Col span={3}>
          <Select
            style={{ width: '100%' }}
            placeholder="Filter by gender"
            value={gender}
            onChange={setGender}
            allowClear
          >
            {GENDERS.map(g => <Select.Option key={g} value={g}>{g}</Select.Option>)}
          </Select>
        </Col>
        <Col span={3}>
          <Checkbox checked={onSale} onChange={handleOnSaleChange}>On Sale</Checkbox>
        </Col>
      </Row>
    </div>
  );

  if (searching) return (
    <Spin size='large' tip="Searching...">{searchBar}</Spin>
  );

  return searchBar;
}

export default Search;
