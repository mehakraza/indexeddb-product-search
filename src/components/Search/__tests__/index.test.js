import React from 'react';
import { Input, Select, Checkbox } from 'antd';
import { shallow } from 'enzyme';
import Search from '..';

const SET_SEARCH_TERM = jest.fn();
const SET_GENDER = jest.fn();
const SET_ON_SALE = jest.fn();

const PROPS = {
   searchRef: {},
   disabled: false,
   setSearchTerm: SET_SEARCH_TERM,
   setGender: SET_GENDER,
   setOnSale: SET_ON_SALE,
   searchTerm: 'searchTerm',
   gender: 'gender',
   onSale: 'onSale',
};

describe('Search', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Search {...PROPS} />);
  });

  describe('rendering', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('callback', () => {
     it('should NOT call setSearchTerm on keypress ENTER', () => {
      wrapper.find(Input).simulate('keypress', { key: 'Enter' });
      expect(SET_SEARCH_TERM.mock.calls).toEqual([]);
    });

    it('should call setSearchTerm on keypress', () => {
      wrapper.find(Input).simulate('keypress', { key: 'j' });
      expect(SET_SEARCH_TERM.mock.calls[0][0]).toEqual('searchTermj');
    });

    it('should call setGender on selection', () => {
      wrapper.find(Select).simulate('change', 'male');
      expect(SET_GENDER.mock.calls[0][0]).toEqual('male');
    });

    it('should call setOnSale on click', () => {
    wrapper.find(Checkbox).simulate('change', {target: { checked: true }});
    expect(SET_ON_SALE.mock.calls[0][0]).toEqual(true);
    });
  });
});
