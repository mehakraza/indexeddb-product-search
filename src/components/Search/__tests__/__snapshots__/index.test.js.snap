// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search rendering should render correctly 1`] = `
<div
  className="search"
>
  <Row
    align="middle"
    gutter={16}
    justify="space-around"
  >
    <Col
      offset={2}
      span={16}
    >
      <Input
        allowClear={true}
        className="search__title"
        disabled={false}
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Search by title"
        type="text"
        value="searchTerm"
      />
    </Col>
    <Col
      span={3}
    >
      <ForwardRef(InternalSelect)
        allowClear={true}
        className="search__gender"
        disabled={false}
        onChange={[MockFunction]}
        placeholder="Filter by gender"
        value="gender"
      >
        <Option
          key="male"
          value="male"
        >
          male
        </Option>
        <Option
          key="female"
          value="female"
        >
          female
        </Option>
        <Option
          key="unisex"
          value="unisex"
        >
          unisex
        </Option>
      </ForwardRef(InternalSelect)>
    </Col>
    <Col
      span={3}
    >
      <Checkbox
        checked="onSale"
        className="search__on-sale"
        disabled={false}
        onChange={[Function]}
      >
        On Sale
      </Checkbox>
    </Col>
  </Row>
</div>
`;
